spec/features/hierarchical_response_display_logic_spec.rb

```diff
@@ -0,0 +1,386 @@
+            {ref_qing_id: qings[:decimal].id, op: "eq", value: "21.72"}
+          ])
+
+        oset = qings[:select_one].option_set
+        qings[:mlev_sel_one] = create_questioning("multilevel_select_one", form,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:select_one].id, op: "eq", option_node_id: oset.node("Dog").id}
+          ])
+
+        oset = qings[:mlev_sel_one].option_set
+        qings[:geo_sel_one] = create_questioning("geo_multilevel_select_one", form,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:mlev_sel_one].id, op: "eq", option_node_id: oset.node("Plant", "Tulip").id}
+          ])
+
+        oset = qings[:geo_sel_one].option_set
+        qings[:select_multiple] = create_questioning("select_multiple", form,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:geo_sel_one].id, op: "eq", option_node_id: oset.node("Canada").id}
+          ])
+
+        oset = qings[:select_multiple].option_set
+        qings[:datetime] = create_questioning("datetime", form,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:select_multiple].id, op: "inc", option_node_id: oset.node("Cat").id}
+          ])
+
+        qings[:date] = create_questioning("date", form,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:datetime].id, op: "lt", value: "#{year}-01-01 5:00:21"}
+          ])
+
+        qings[:time] = create_questioning("time", form,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:date].id, op: "eq", value: "#{year}-03-22"}
+          ])
+
+        qings[:text3] = create_questioning("text", form,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:time].id, op: "geq", value: "3:00pm"}
+          ])
+
+        qings[:grp1] = create_questioning("text", form, parent: group)
+
+        qings[:rpt1] = create_questioning("text", form,
+          parent: rpt_group,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:text3].id, op: "eq", value: "baz"} # References top level Q
+          ])
+
+        qings[:rpt2] = create_questioning("text", form,
+          parent: rpt_group,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:rpt1].id, op: "eq", value: "qux"} # References same group Q
+          ])
+
+        qings[:rpt3] = create_questioning("text", form,
+          parent: rpt_group,
+          display_if: "all_met",
+          display_conditions_attributes: [
+            {ref_qing_id: qings[:grp1].id, op: "eq", value: "nix"} # References Q from sibling group
+          ])
```

Comment:
Perhaps we could add a nested group in here without too much trouble? And add in the bit that tests that we search first in our own neighborhood before globally?